modules/security-jwt/src/main/java/com/devonfw/module/security/jwt/sign/RsaJwtSignatureAlgorithmImpl.java

```diff
@@ -0,0 +1,48 @@
+package com.devonfw.module.security.jwt.sign;
+
+import java.security.interfaces.RSAPrivateKey;
+import java.security.interfaces.RSAPublicKey;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import org.springframework.security.jwt.crypto.sign.RsaSigner;
+import org.springframework.security.jwt.crypto.sign.RsaVerifier;
+import org.springframework.security.jwt.crypto.sign.SignatureVerifier;
+import org.springframework.security.jwt.crypto.sign.Signer;
+
+import com.devonfw.module.security.keystore.config.KeyStoreAccess;
+
+/**
+ * Implementation of {@link JwtSignatureAlgorithm}
+ *
+ * @since 3.2.0
+ *
+ */
+@Named
+public class RsaJwtSignatureAlgorithmImpl implements JwtSignatureAlgorithm {
+
+  private final static String ALGORITHM = "RSA";
+
+  @Inject
+  private KeyStoreAccess keyStoreAccess;
+
+  @Override
+  public String getName() {
+
+    return ALGORITHM;
+  }
+
+  @Override
+  public Signer createSigner() {
+
+    return new RsaSigner((RSAPrivateKey) this.keyStoreAccess.getPrivateKey());
```

Comment:
We need to be careful. When is this code used? Most apps will not issue JWTs and do not have a private key. Therefore such configuration will not be present and this method would currently end up in a runtime exception.